lib/truffle/rbconfig.rb

```diff
@@ -214,9 +219,11 @@ module RbConfig
     'RMDIR'             => rmdir,
     'RMDIRS'            => "#{rmdir} -p",
     'RPATHFLAG'         => ' -Wl,-rpath,%1$-s',
+    'RUBY_API_VERSION'  => "#{major}.#{minor}",
     'RUBY_BASE_NAME'    => ruby_base_name,
     'ruby_install_name' => ruby_install_name,
     'RUBY_INSTALL_NAME' => ruby_install_name,
+    'RUBY_PROGRAM_VERSION' => RUBY_VERSION.dup,
```

Comment:
Not sure what's this key is supposed to be e.g. RUBY_VERSION or RUBY_ENGINE_VERSION. (JRuby doesn't have it FWIW).

https://github.com/ruby/ruby/blob/74f315c3f5908512dfdeeab2dd76d93e98149769/configure.ac#L126-L130 that's how it's defined on CRuby.
And the fact they are defined like that might mean they are customizable? (unsure)

But not sure what it should be.
I would tend to say `RUBY_ENGINE_VERSION.dup` is safer here if it's the version of the `ruby` program/executable.

Reply:
Good question.
It doesn't look like `rb-sys` cares what the value is (just, if it changes, presumably).

With a [github code search](https://github.com/search?q=%22RUBY_PROGRAM_VERSION%22+-is%3Afork&type=code&p=1) I do see several examples where there is an expectation that `RUBY_PROGRAM_VERSION` matches `MAJOR`, `MINOR`, etc:

This is super old, but is using it to match the "ruby" version:
https://github.com/rocky/rb-trepanning/blob/dab44a65144dcad3b9529bc1b8df6b3683f61ec2/Rakefile#L17-L29
Similar, just making sure the running ruby is compatible enough:
https://github.com/farmisen/her/blob/2ed62ccf4fac16cd54d2082a9563749be10ab945/Gemfile#L4-L9

Another like `rb-sys`: https://github.com/vericred/rutie/blob/d2034202a0397afe84b0bd282f7828b5a388bd94/build.rs#L81

This is an interesting one... not really sure what the repo is for but MAJOR and MINOR are inferred from RUBY_PROGRAM_VERSION:
https://github.com/nobu/build-files/blob/4f98f5dca24e9de97432b7d71dbbfaddef3164da/Ruby.mk#L131-L133

Another assumption that this matches the "ruby version" rather than engine 
`  s.required_ruby_version = c['RUBY_PROGRAM_VERSION']`
https://github.com/rednovalabs/rubyuno/blob/3c53b9d48d6e9d8583fe3c1f4fa748ff016b8adc/Rakefile#L328

This one creates "language version" with it
`            language_version = "#{rbconfig["RUBY_PROGRAM_VERSION"]}-p#{rbconfig["PATCHLEVEL"]}"`
https://github.com/appsignal/appsignal-ruby/blob/aa51c02d34c26a2120396064c0c9967e818ee0cc/lib/appsignal/cli/diagnose.rb#L446

So in the end, I'm still not sure.  It seems like people expect it to be the "language" version, but I'm not sure if it would cause problems being the engine version, and could be more accurate for some reporting 🤷